physionet-django/project/models.py

```diff
@@ -675,6 +689,169 @@ def get_anonymous_url(self):
 
         return anonymous.url
 
+    def citation_text(self, styles):
+        """
+        Citation information in multiple formats (MLA, APA, Chicago,
+        Harvard, and Vancouver).
+
+        1. MLA (8th edition) [https://owl.purdue.edu/owl/research_and_citation/
+                              mla_style/mla_formatting_and_style_guide/
+                              mla_formatting_and_style_guide.html]
+        2. APA (7th edition) [https://owl.purdue.edu/owl/research_and_citation/
+                              apa_style/apa_style_introduction.html]
+        3. Chicago (17th edition) [https://owl.purdue.edu/owl/
+                                   research_and_citation/
+                                   chicago_manual_17th_edition/
+                                   cmos_formatting_and_style_guide/
+                                   chicago_manual_of_style_17th_edition.html]
+        4. Harvard [https://www.mendeley.com/guides/harvard-citation-guide]
+        5. Vancouver [https://guides.lib.monash.edu/ld.php?content_id=14570618]
+
+        Parameters
+        ----------
+        styles [list]:
+            ['MLA', 'APA', 'Chicago', 'Harvard', 'Vancouver']
+
+        Returns
+        -------
+        citation_dict [dict]:
+            dictionary containing all desired citation styles
+        """
+        authors = self.authors.all().order_by('display_order')
+        citation_dict = {}
+
+        if self.is_published():
+
+            year = self.publish_datetime.year
+            doi = self.doi
+            is_legacy = models.BooleanField(default=False)
+
+            if self.is_legacy:
+                return {'APA': ''}
+
+        else:
+            """
+            Since the project is not yet published, the current year is used in
+            place of the publication year, and '*****' is used in place of the
+            DOI suffix.
+            """
+            year = timezone.now().year
+            doi = '10.13026/*****'
+
+        shared_content = {'year': year,
+                          'title': self.title,
+                          'version': self.version}
+
+        for style in styles:
+
+            if style == 'MLA':
+
+                style_format = ('{author}. "{title}" (version {version}). '
+                                '<i>PhysioNet</i> ({year})')
+
+                doi_format = (', <a href="https://doi.org/{doi}">'
+                              'https://doi.org/{doi}</a>.')
+
+                if (len(authors) == 1):
+                    all_authors = authors[0].get_full_name(reverse=True)
+                elif (len(authors) == 2):
+                    first_author = authors[0].get_full_name(reverse=True)
+                    second_author = authors[1].get_full_name()
```

Comment:
Should this be reversed too?

Reply:
Actually no, the style guide want the final name to be written in normal order like "Pollard T., Lucas McCullum" or something similar to that.